File: syspzl-api/src/repositories/veiculo.repository.js
import pool from '../database/database.connection.js';

class VeiculoRepository {
  // Busca todos os veículos de um autuado específico
  async findByAutuadoId(autuadoId) {
    const [rows] = await pool.query('SELECT * FROM `veiculo` WHERE autuado_id = ?', [autuadoId]);
    return rows;
  }

  async findById(id) {
    const [rows] = await pool.query('SELECT * FROM `veiculo` WHERE id = ?', [id]);
    return rows[0];
  }

  async findByPlaca(placa) {
    const [rows] = await pool.query('SELECT * FROM `veiculo` WHERE placa = ?', [placa]);
    return rows[0];
  }

  async create(veiculoData) {
    const { autuado_id, placa, marca_modelo } = veiculoData;
    const [result] = await pool.query(
      'INSERT INTO `veiculo` (autuado_id, placa, marca_modelo) VALUES (?, ?, ?)',
      [autuado_id, placa, marca_modelo]
    );
    return { id: result.insertId, ...veiculoData };
  }

  async update(id, veiculoData) {
    const { placa, marca_modelo } = veiculoData;
    const [result] = await pool.query(
      'UPDATE `veiculo` SET placa = ?, marca_modelo = ? WHERE id = ?',
      [placa, marca_modelo, id]
    );
    return result;
  }

  async delete(id) {
    const [result] = await pool.query('DELETE FROM `veiculo` WHERE id = ?', [id]);
    return result;
  }
}

export default new VeiculoRepository();

File: api-python/src/persistence/veiculo_repository.py
from mysql.connector import Error

from src.config.connection.connection import DatabaseConnection
from src.schemas.veiculo import VeiculoCreate

class VeiculoRepository:
    def buscar_todos(self):
        with DatabaseConnection() as cnx:
            try:
                query = "SELECT * FROM veiculo"
                cnx.execute_query(query)
                return cnx.cursor.fetchall()
            except Error as e:
                raise(e)
            
    def buscar_por_id(self, id:int):
        with DatabaseConnection() as cnx:
            try:
                query = "SELECT * FROM veiculo WHERE id = %s"
                cnx.execute_query(query, [id])
                return cnx.cursor.fetchall()
            except Error as e:
                raise(e)

    def criar(self, veiculo:VeiculoCreate):
        with DatabaseConnection() as cnx:
            try:
                params = [veiculo.autuado_id, veiculo.cor, veiculo.marca_modelo, veiculo.placa]
                query = "INSERT INTO veiculo (autuado_id, cor, marca_modelo, placa) VALUES (%s, %s, %s, %s)"
                cnx.execute_query(query, params)
                cnx.connection.commit()
                return cnx.cursor.lastrowid
            except Error as e:
                raise(e)
            
    def deletar(self, id:int):
        with DatabaseConnection() as cnx:
            try:
                query = "DELETE FROM veiculo WHERE id = %s"
                cnx.execute_query(query, [id])
                cnx.connection.commit()
                return cnx.cursor.rowcount
            except Error as e:
                raise(e)